### PR TITLE
Various optimisations

### PR DESCRIPTION
**Pull request by stefanceriu**
* Cache TimelineBubbleLayout subview sizes
* Cache MessageText sizes, avoid extra updates
* Only use the `CollapsibleReactionLayout` if there's more than 5 reactions on a particular message
* Upgrade Sentry to 8.13.0, disable various options as they're not useful and impact performance

### CODE REVIEW

File: ElementX/Sources/Services/BugReport/BugReportService.swift [hunk 1]
```diff
@@ -69,6 +69,14 @@ class BugReportService: NSObject, BugReportServiceProtocol {
             
             options.dsn = self.sentryURL.absoluteString
             
+            // Sentry swizzling shows up quite often as the heaviest stack trace when profiling
+            // We don't need any of the features it powers (see docs)
+            options.enableSwizzling = false
+            
+            // WatchdogTermination is currently the top issue but we've had zero complaints
+            // so it might very well just all be false positives
+            options.enableWatchdogTerminationTracking = false
```
Comment (pixlwave) [inline ElementX/Sources/Services/BugReport/BugReportService.swift:78]: Just a thought: Did you check the release notes for the newer versions to see if there were any fixed bugs around this?

File: ElementX/Sources/Screens/RoomScreen/View/Style/TimelineBubbleLayout.swift [hunk 2]
```diff
@@ -36,30 +40,41 @@ struct TimelineBubbleLayout: Layout {
...
     
-    func placeSubviews(in bounds: CGRect, proposal: ProposedViewSize, subviews: Subviews, cache: inout ()) {
+    func placeSubviews(in bounds: CGRect, proposal: ProposedViewSize, subviews: Subviews, cache: inout Cache) {
         guard !subviews.isEmpty else { return }
         
         // Calculate the width using the regular text and the non-greedy quote bubbles.
         let layoutSubviews = subviews.filter { $0.priority != Priority.visibleQuote }
-        let maxWidth = layoutSubviews.map { $0.sizeThatFits(proposal).width }.reduce(0, max)
         
-        // Place the regular text and greedy quote bubbles using the calculated width.
```
Comment (pixlwave) [inline ElementX/Sources/Screens/RoomScreen/View/Style/TimelineBubbleLayout.swift:60]: Was this removal necessary? 😢

File: ElementX/Sources/Screens/RoomScreen/View/Style/TimelineBubbleLayout.swift [hunk 3]
```diff
@@ -70,4 +85,25 @@ struct TimelineBubbleLayout: Layout {
             y += height + spacing
         }
     }
+    
+    // MARK: - Private
+    
+    private func cachedSizeFor(subview: LayoutSubview, subviews: LayoutSubviews, proposedSize: ProposedViewSize, cache: inout Cache) -> CGSize {
```
Comment (pixlwave) [inline ElementX/Sources/Screens/RoomScreen/View/Style/TimelineBubbleLayout.swift:91]: This name is slightly confusing at the call site as it implies the size *only* comes from the cache. Its not a great suggestion, but `sizeThatFitsUsingCache` or something would be clearer.

File: ElementX/Sources/Screens/RoomScreen/View/Supplementary/TimelineReactionsView.swift [hunk 4]
```diff
@@ -34,6 +34,40 @@ struct TimelineReactionsView: View {
...
+            Button {
+                context.send(viewAction: .displayEmojiPicker(itemID: itemID))
+            } label: {
+                TimelineReactionAddMoreButtonLabel()
+            }
+        }
+        .environment(\.layoutDirection, reactionsLayoutDirection)
+        .animation(.easeInOut(duration: 0.1).disabledDuringTests(), value: reactions)
+        .padding(.leading, 4)
+    }
```
Comment (pixlwave) [inline ElementX/Sources/Screens/RoomScreen/View/Supplementary/TimelineReactionsView.swift:68]: A neater solution would be to use `AnyLayout` in the collapsibleBody so we don't duplicate any code.

**Review by pixlwave — APPROVED** [review]
I'm not massively convinced reverting to an `HStack` is better than figuring out the Cache in the reactions layout, but the code lgtm.

File: ElementX/Sources/Screens/RoomScreen/View/Style/TimelineBubbleLayout.swift (hunk 2 above)
Comment (stefanceriu) [reply ElementX/Sources/Screens/RoomScreen/View/Style/TimelineBubbleLayout.swift:60]: Oh, sorry, no, complete mistake 👍

File: ElementX/Sources/Services/BugReport/BugReportService.swift (hunk 1 above)
Comment (stefanceriu) [reply ElementX/Sources/Services/BugReport/BugReportService.swift:78]: I did yes, nothing for that. The only interesting bit was `Symbolicate locally only when debug is enabled` in `8.9.0`

File: ElementX/Sources/Screens/RoomScreen/View/Style/TimelineBubbleLayout.swift (hunk 3 above)
Comment (stefanceriu) [reply ElementX/Sources/Screens/RoomScreen/View/Style/TimelineBubbleLayout.swift:91]: How about just `sizeFor`? :D

Comment (pixlwave) [reply ElementX/Sources/Screens/RoomScreen/View/Style/TimelineBubbleLayout.swift:91]: How about `size(for:)` then 😉
